### PR TITLE
fix(forge/install): check github repo name is valid

### PR DESCRIPTION
**Pull request by jrhe**
## Motivation

`forge install` doesn't check to see if a valid github repo name has been specified when using the shorthand method for installing packages e.g. `forge install solmate`.

See #712.

## Solution

Adds a (basic) regex to check the format of the github repo name, and throws an error if invalid.

### CODE REVIEW

**Review by mattsse — CHANGES_REQUESTED** [review]
thanks for this

File: cli/src/opts/forge.rs [hunk 1]
```diff
@@ -206,6 +209,13 @@ impl FromStr for Dependency {
         } else if dependency.starts_with(GITHUB) {
             format!("https://{}", dependency)
         } else {
+            lazy_static! {
```
[the remark this answers is not in the document]
Comment (tdudz) [reply cli/src/opts/forge.rs:212]: just curious, why is `once_cell` considered better?

[the remark this answers is not in the document]
Comment (gakonst) [reply cli/src/opts/forge.rs:212]: https://users.rust-lang.org/t/lazy-static-vs-once-cell-oncecell/58578/3

**Comment by sambacha** [comment]
🙏💯

thanks

**Pull request merged**